assets/src/design-system/components/typeahead/typeahead.js

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { THEME_CONSTANTS } from '../../theme';
+import { Menu, MENU_OPTIONS } from '../menu';
+import { Popup, PLACEMENT } from '../popup';
+import { DropDownContainer, Hint } from './components';
+import {
+  DEFAULT_POPUP_FILL_WIDTH,
+  DEFAULT_PLACEHOLDER,
+  DEFAULT_ARIA_CLEAR_LABEL,
+  DEFAULT_ARIA_INPUT_LABEL,
+} from './constants';
+import TypeaheadInput from './input';
+import useTypeahead from './useTypeahead';
+
+/**
+ *
+ * @param {Object} props All props.
+ * @param {string} props.ariaInputLabel Aria label passed to input to label for SR users since this component has no visible label.
+ * @param {string} props.ariaClearLabel Aria label passed to clear button for SR users.
+ * @param {boolean} props.disabled If true, menu will not be openable
+ * @param {string} props.emptyText If the array of options is empty this text will display when menu is expanded.
+ * @param {Function} props.handleTypeaheadValueChange specific callback to monitor changes in input value separate from onMenuItemClick.
+ * This is to separate the inputValue from the selectedValue should the typeahead demand selection from a list as opposed to be freeform.
+ * @param {boolean} props.hasError If true, input and hint (if present) will show error styles.
+ * @param {string} props.hint Hint text to display below input (optional). If not present, no hint text will display.
+ * @param {boolean} props.isFlexibleValue If true, input values will be treated same as options on selected on enter and clearing the input will also clear the selectedValue.
```

Comment:
i'm a little confused by this description... you're saying the selection handler functions will return whatever the input value is when this is true?

Reply:
Updated the wording